src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1056,11 +1057,13 @@ private function createService(Definition $definition, $id, $tryProxy = true)
                 }
             }
         } else {
-            $r = new \ReflectionClass($parameterBag->resolveValue($definition->getClass()));
+            $r = new \ReflectionClass($class = $parameterBag->resolveValue($definition->getClass()));
 
             $service = null === $r->getConstructor() ? $r->newInstance() : $r->newInstanceArgs($arguments);
+             // don't trigger deprecations for internal uses, to be removed in 4.0 along with the deprecated class
```

Comment:
personally, I try to always add the `@deprecated` string to ease doing grep. I don't expect anyone to read all the source code looking for specific notices about 4.0 (or we'd expect to miss some for sure)